gufe/settings/models.py

```diff
@@ -0,0 +1,174 @@
+"""
+Pydantic models used for storing settings.
+"""
+
+from datetime import date
+from typing import Literal, Union
+
+from pydantic import BaseModel, Extra, FilePath, PositiveFloat
+from typing_extensions import TypedDict
+
+
+class SettingsBaseModel(BaseModel):
+    """
+    Settings and modifications we want for all settings classes.
+    """
+
+    class Config:
+        extra = Extra.forbid
+        # Immutability in python is never strict.
+        # If developers are determined/stupid
+        # they can always modify a so-called "immutable" object.
+        # https://pydantic-docs.helpmanual.io/usage/models/#faux-immutability
+        allow_mutation = False
+        arbitrary_types_allowed = True
+        smart_union = True
+
+
+class vdWScale(TypedDict):
+    scale12: float
+    scale13: float
+    scale14: float
+    scale15: float
+
+
+class ElectrostaticScale(TypedDict):
+    scale12: float
+    scale13: float
+    scale14: float
+    scale15: float
+
+
+class VdWSettings(SettingsBaseModel):
+    """
+    Settings for van der Waals force
+    """
+
+    combining_rules: Literal["Lorentz-Berthelot"]
+    potential: Literal["Lennard-Jones-12-6"]
+    scale: vdWScale
+    long_range_dispersion: Literal["isotropic"]
+    cutoff: float  # angstrom
```

Comment:
Sorry I was less fancy in this comment than I was in others, here we would require a unit with dimension of length. We can then convert it to whatever unit the tool needs when we pass it in.

Reply:
We really only care about the dimension of the quantity users us give us since we can always convert. We could be more strcit but honestly I don't care if they pass us `9 * unit.angstrom` or `0.9 * unit.nm`

Reply:
Yeah the nice thing about units is you can use any unit that is in the correct dimension(s)